yb-voyager/src/query/queryissue/detectors.go

```diff
@@ -31,86 +31,115 @@ const (
 // To Add a new unsupported query construct implement this interface for all possible nodes for that construct
 // each detector will work on specific type of node
 type UnsupportedConstructDetector interface {
-	Detect(msg protoreflect.Message) ([]string, error)
+	Detect(msg protoreflect.Message) error
+	GetIssues() []QueryIssue
 }
 
 type FuncCallDetector struct {
+	query string
 	// right now it covers Advisory Locks and XML functions
-	unsupportedFuncs map[string]string
+	advisoryLocksFuncsDetected mapset.Set[string]
+	xmlFuncsDetected           mapset.Set[string]
```

Comment:
great, now we are storing info about what all functions of advisory lock or xml are detected, can be used in future.